jax/numpy/lax_numpy.py

```diff
@@ -158,9 +166,16 @@ def _promote_shapes(*args):
     return args
   else:
     shapes = [shape(arg) for arg in args]
-    nd = len(lax.broadcast_shapes(*shapes))
-    return [lax.reshape(arg, (1,) * (nd - len(shp)) + shp)
-            if shp and len(shp) != nd else arg for arg, shp in zip(args, shapes)]
+    if len(set(shapes)) == 1:
+      return args
+    elif FLAGS.jax_numpy_rank_promotion:
+      nd = len(lax.broadcast_shapes(*shapes))
+      return [lax.reshape(x, (1,) * (nd - len(s)) + s)
+              if s and len(s) != nd else x for x, s in zip(args, shapes)]
```

Comment:
shp and arg were clearer than s and x...

Reply:
Done.